src/main/java/com/github/romankh3/image/comparison/ImageComparisonUtil.java

```diff
@@ -99,4 +95,90 @@ public static void saveImage(File pathFile, BufferedImage image) throws IOExcept
         }
         ImageIO.write(image, "png", pathFile);
     }
+
+    /**
+     * Resize image to new dimensions and return new image
+     *
+     * @param img the object of the image to be resized.
+     * @param newW the new width.
+     * @param newH the new height.
+     */
+
+    public static BufferedImage resize(BufferedImage img, int newW, int newH)  {
+        Image imgtmp = img;
+
+
+        BufferedImage newImage = toBufferedImage( imgtmp.getScaledInstance(newW, newH, Image.SCALE_SMOOTH));
+        return newImage;
+    }
+
+    /**
+     * convert image to Buffered Image.
+     *
+     * @param img the object of Image to be converted.
+     */
+
+    public static BufferedImage toBufferedImage(Image img)
+    {
+        if (img instanceof BufferedImage)
+        {
+            return (BufferedImage) img;
+        }
+
+        float softenFactor = 0.05f;
+        final Image temp = new ImageIcon(img).getImage();
+        final BufferedImage bufferedImage = new BufferedImage(
+                temp.getWidth(null),
+                temp.getHeight(null),
+                BufferedImage.TYPE_INT_RGB);
+        final Graphics g = bufferedImage.createGraphics();
+        g.setColor(Color.white);
+        g.fillRect(0, 0, temp.getWidth(null), temp.getHeight(null));
+        g.drawImage(temp, 0, 0, null);
+        g.dispose();
+
+        final float[] softenArray = {0, softenFactor, 0, softenFactor, 1 - (softenFactor * 4), softenFactor, 0, softenFactor, 0};
+        final Kernel kernel = new Kernel(3, 3, softenArray);
+        final ConvolveOp cOp = new ConvolveOp(kernel, ConvolveOp.EDGE_NO_OP, null);
+        final BufferedImage filteredBufferedImage = cOp.filter(bufferedImage, null);
+
+        return filteredBufferedImage;
+    }
+
+    /**
+     * Return difference percent between two buffered images.
+     * @param img1 the first image.
+     * @param img2 the second image.
+     * @return difference percent.
+     */
+    public static float getDifferencePercent(BufferedImage img1, BufferedImage img2) {
+        int width = img1.getWidth();
+        int height = img1.getHeight();
+
+        long diff = 0;
+        for (int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {
+                diff += pixelDiff(img1.getRGB(x, y), img2.getRGB(x, y));
+            }
+        }
+        long maxDiff = 3L * 255 * width * height;
+
+        return (float) (100.0 * diff / maxDiff);
+    }
+
+    /**
+     * Compare to pixels
+     * @param rgb1 the first rgb
+     * @param rgb2 the second rgn
+     * @return the difference.
+     */
+    public static int pixelDiff(int rgb1, int rgb2) {
```

Comment:
the method almost the same exists in the project. We need to avoid duplicate code. Please, provide the solution to avoid it.